src/ast.c

```diff
@@ -326,9 +326,14 @@ static jl_value_t *scm_to_julia_(value_t e, int eo)
 
             e = cdr_(e);
             if (!eo) {
-                if (sym == line_sym && n==1) {
-                    return jl_new_struct(jl_linenumbernode_type,
-                                         scm_to_julia_(car_(e),0));
+                if (sym == line_sym && n==2) {
+                    jl_value_t *filename = scm_to_julia_(car_(cdr_(e)),0);
+                    jl_value_t *linenum  = scm_to_julia_(car_(e),0);
+                    JL_GC_PUSH2(&filename, &linenum);
```

Comment:
note that assuming filename is a Symbol, it doesn't need a root. If it wasn't a symbol, this root would have been needed before the call to create the box to linenum.